app/payment-link-v2/reference/reference.njk

```diff
@@ -0,0 +1,59 @@
+{% extends "../../views/layout.njk" %}
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% from "govuk/components/input/macro.njk" import govukInput %}
+{% from "govuk/components/back-link/macro.njk" import govukBackLink %}
+{% from "../../views/macros/error-summary.njk" import errorSummary %}
+
+{% block pageTitle %}
+  {{ __p('paymentLinksV2.amount.pleaseEnterYour') }} {{product.reference_label}}  - {{ productName }}
+{% endblock %}
+
+{% block contentBody %}
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    {{ govukBackLink({
+      text: "Back",
+      href: backLinkHref
+    }) }}
+
+    {{ errorSummary ({
+      errors: errors,
+      hrefs: {
+        'payment-reference': '#payment-reference'
+      }
+    }) }}
+
+    <form action="/pay/{{ productExternalId }}/reference" method="post" name="startPaymentForm" class="push-bottom">
+      <input id="csrf" name="csrfToken" type="hidden" value="{{ csrf }}"/>
+
+      {% set paymentAmountError = false %}
+      {% if errors['payment-reference'] %}
+        {% set paymentAmountError = {
+          text: errors['payment-reference']
+        } %}
+      {% endif %}
+
+      {{ govukInput({
+        id: "payment-reference",
+        name: "payment-reference",
+        errorMessage: paymentAmountError,
```

Comment:
a very minor thing, but I like to inline the error messages to save all the boilerplate in the template. Like this:

```
errorMessage: { text: errors['payment-reference']} if errors.['payment-reference'] else false
```

Reply:
Good idea - updated the code.